src/Conversion/KrnlToLLVM/KrnlUnaryMath.cpp

```diff
@@ -146,22 +146,30 @@ class KrnlUnaryMathOpLowering : public ConversionPattern {
 
     // get the LLVM type for the function args and result
     mlir::Type inType = op->getOperand(0).getType();
-    mlir::Type llvmType;
+    mlir::Type outType = op->getResultTypes().front();
+    mlir::Type llvmInType, llvmOutType;
     if (inType.isF32())
-      llvmType = FloatType::getF32(context);
+      llvmInType = FloatType::getF32(context);
     else if (inType.isF64())
-      llvmType = FloatType::getF64(context);
+      llvmInType = FloatType::getF64(context);
+    if (outType.isInteger(1))
+      llvmOutType = IntegerType::get(context, 1);
+    else if (outType.isF32())
+      llvmOutType = FloatType::getF32(context);
+    else if (outType.isF64())
+      llvmOutType = FloatType::getF64(context);
```

Comment:
Are those code intended to convert MLIR type to llvm type? Can llvm type converter be used?

Reply:
@hamptonm1 I assume that this is code that you got from an earlier PR. 
@chentong319 can you point to code that use the llvm type converter?

Reply:
Yeah this is what was previously there